tests/suites/test_suite_pk.data

```diff
@@ -13,135 +13,135 @@ depends_on:MBEDTLS_RSA_C:MBEDTLS_GENPRIME
 pk_utils:MBEDTLS_PK_RSA:MBEDTLS_RSA_GEN_KEY_MIN_BITS:MBEDTLS_RSA_GEN_KEY_MIN_BITS:(MBEDTLS_RSA_GEN_KEY_MIN_BITS /8):"RSA"
 
 PK utils: ECKEY SECP192R1
-depends_on:MBEDTLS_PK_HAVE_ECC_KEYS:MBEDTLS_ECP_DP_SECP192R1_ENABLED
+depends_on:MBEDTLS_PK_HAVE_ECC_KEYS:MBEDTLS_ECP_HAVE_SECP192R1
```

Comment:
This test case fails in `all.sh test_psa_crypto_config_accel_ecc_non_weirstrass_curves` in 5c21036d19ff922b65a7fc952f5ea17dafbc3aa5.

If some curves are accelerated but not others, and `MBEDTLS_USE_PSA_CRYPTO` is enabled, we [don't define `MBEDTLS_PK_USE_PSA_EC_DATA`](https://github.com/Mbed-TLS/mbedtls/blob/87fe99627f30983dfdbf7fba0506464853cbf0d2/include/mbedtls/pk.h#L174) (makes sense: that's only for a pure-PSA world, not for a mixed world). Then the code in `test_suite_pk` uses the [legacy code path to generate an EC key](https://github.com/Mbed-TLS/mbedtls/blob/87fe99627f30983dfdbf7fba0506464853cbf0d2/tests/suites/test_suite_pk.function#L104). That's a bug in the test code: nowadays it should base the choice of code path on where the support is for the specific curve, or it could dispatch to PSA whenever PSA is available. I think this violates the objective of https://github.com/Mbed-TLS/mbedtls/pull/8075.

I'm not sure what the best resolution is at this point, given that this PR is already very long and we want to release ASAP. Do we have to fix it now? Or should we tone down the testing here and file it as a bug to be fixed after the release?

Reply:
As much as releasing something with a known bug is rather unpleasant, I am inclined to say it's probably the best option here. My vote goes to toning down the testing and fixing with a follow-up PR post release.

Reply:
Let's spend a bit more time investigating this and seeing if we can get to a point where we can get the testing in good shape. Slipping a few days while we explore this more isn't going to be a big issue, rather that than risk a buggy release.

Reply:
I'm inclined to temporarily tone done the testing here and merge, then investigate and fix in a follow-up, just to reduce the risk of conflicts / complexity of rebasing, as well as to unblock #8242 which depends on this one.

Then depending on how the follow-up is going, we can see what the impact on the release is. Wdyt?

Reply:
I am in favour of this approach too, personally.

Reply:
That sounds reasonable to me.

Reply:
Fine with me. That, or keep the offending tests disabled when the curve is accelerated: as far as I can tell, they assume that the key is not opaque (`mbedtls_pk_get_type(&pk) == type`).

I think the tests are attempting to do something reasonable, which isn't working when some curves are accelerated. So I think this indicates that we haven't achieved proper support for builds where some curves are accelerated and others are not. But it's possible that the tests are just overly picky and need to be rewritten, and it's just ok that switching curves to be accelerated is not a transparent change for users of the pk module.

Reply:
My reasoning: starting at 5c21036d19ff922b65a7fc952f5ea17dafbc3aa5, I attempted to make this test pass in `all.sh test_psa_crypto_config_accel_ecc_non_weirstrass_curves`. The test needs to generate an EC key on a curve which may or may not be accelerated. If the curve is not accelerated then `MBEDTLS_ECP_C` is enabled and the `mbedtls_ecp_gen_keypair` code path is taken, which is fine. If all curves are accelerated then `MBEDTLS_PK_USE_PSA_EC_DATA` is enabled, and all is fine. But if this curve is accelerated while others are not, we have a problem. Since `MBEDTLS_ECP_C` is present (for other curves), `MBEDTLS_PK_USE_PSA_EC_DATA` is disabled. But `mbedtls_ecp_gen_keypair` and even `mbedtls_ecp_group_load` won't work for the accelerated curve.

I am worried that we are breaking something that worked in 3.4, however. So *I don't want to just disable the test components*. I'd like to restore the test components that were removed (`
component_test_psa_crypto_config_accel_all_curves_except_p192` and `component_test_psa_crypto_config_accel_all_curves_except_x25519`). If those don't work, that's a regression and I would not approve this pull request.

Reply:
Note: there was no `component_test_psa_crypto_config_accel_all_curves_except_p192` (or similar) in 3.4.

Also, this entire PR is fixing an API bug, that was present in 3.4: before this PR, you could declare algs and key types as accelerated but declare no curve as accelerated, and still have your build be driver-only - it was implicitly assumed that curves were accelerated. So, we are breaking things that used to work in 3.4, but that's on purpose, because those should never have worked in the first place. (Now that I'm writing this, I realise we want a ChangeLog entry under "Changes" here.)

> I would much rather sort out some technical debt (in the bridge between pk and psa, and in how we do builds with drivers) before sorting out the problems here. Hence my preference for just shipping what we have (which is, at the very least, useful for builds where everything ECC is accelerated).

I tend to agree that we care mostly about the case where all required ECC is accelerated, and before spending too much time on supporting partially-accelerated we need to think about our priorities.

Of course, if we decide to only support the case where all ECC is accelerated, we should explicitly say so in the ChangeLog and documentation. (Actually, I think that's a bit overkill: we do support the case where some algs are accelerated but not others, what we don't seem to support easily is the case where come curves are accelerated but not all of them.)

Reply:
> there was no component_test_psa_crypto_config_accel_all_curves_except_p192 (or similar) in 3.4.

Ah, ok. So this is (I suspect) a regression compared to development, but not compared to the last release. Then I'm ok with:

* removing the new components that are failing;
* filing an issue describing the problem;
* documenting that we support all-curves-accelerated, but support for some-curves-accelerated is untested and known not to work in all cases.

Reply:
"some-curves-accelerated is untested and known not to work in all cases" - presumably the case where only SECP256R1 is accelerated is tested and known to work?

Reply:
> the case where only SECP256R1 is accelerated is tested

No, we don't test this anywhere, as far as I can tell. If we did, it would fail for the same reason as `test_psa_crypto_config_accel_ecc_weirstrass_curves` here. In particular, we only test P256-m in the TF-M config, and it only enables that single curve.

Reply:
Ah, I see - sorry I misunderstood. We support the case where all curves are accelerated, including when that is just one curve: so we support the TF-M use-case.

Reply:
> * removing the new components that are failing;
> * documenting that we support all-curves-accelerated, but support for some-curves-accelerated is untested and known not to work in all cases.

Done in [f7dc6cf](https://github.com/Mbed-TLS/mbedtls/pull/8075/commits/f7dc6cfef18627459e2b151c1d1191167ea1c9a6) except I though removing the whole components was a bit heavy-handed: currently we got all of PSA working in those components, that's already something, so I decided to keep it tested and document it as working. Excluded from tests and documented as "not working": PK (because tests fail), and X.509+TLS (because they use PK so it feels wrong to document them as working when PK is not).

> * filing an issue describing the problem;

https://github.com/Mbed-TLS/mbedtls/issues/8255